dbms/programs/server/PrometheusMetricsWriter.cpp

```diff
@@ -96,6 +98,39 @@ void PrometheusMetricsWriter::write(WriteBuffer & wb) const
             writeOutLine(wb, key, value);
         }
     }
+
+    if (send_status_info)
+    {
+        for (size_t i = 0, end = CurrentStatusInfo::end(); i < end; ++i)
+        {
+            std::lock_guard<std::mutex> lock(CurrentStatusInfo::locks[static_cast<CurrentStatusInfo::Status>(i)]);
+            std::string metric_name{CurrentStatusInfo::getName(static_cast<CurrentStatusInfo::Status>(i))};
+            std::string metric_doc{CurrentStatusInfo::getDocumentation(static_cast<CurrentStatusInfo::Status>(i))};
+
+            replaceInvalidChars(metric_name);
+            std::string key{current_status_prefix + metric_name};
+
+            writeOutLine(wb, "# HELP", key, metric_doc);
+            writeOutLine(wb, "# TYPE", key, "gauge");
+
+            for (const auto & value: CurrentStatusInfo::values[i])
+            {
+                for (const auto & enum_value: CurrentStatusInfo::getAllPossibleValues(static_cast<CurrentStatusInfo::Status>(i)))
+                {
+                    DB::writeText(key, wb);
+                    DB::writeChar('{', wb);
+                    DB::writeText(key, wb);
+                    DB::writeText("=\"", wb);
+                    DB::writeText(enum_value.first, wb);
+                    DB::writeText("\",name=\"", wb);
+                    DB::writeText(value.first, wb);
```

Comment:
shouldn't this also has `replaceInvalidChars`, since the dictionary name can has database in it's name (AFAICS it uses full name, and indeed it should)

Reply:
Hum I guess maybe we should escape double-quote characters, but as it's a string, it should be ok in theory ? maybe I could be wrong on this one

Reply:
According to [prometheus docs](https://prometheus.io/docs/concepts/data_model/):

>Label names may contain ASCII letters, numbers, as well as underscores. They must match the regex [a-zA-Z_][a-zA-Z0-9_]*. Label names beginning with __ are reserved for internal use.

IOW there is no `.` which can be in the full dictionary name

Reply:
Egh, sorry, this is names not values